src/cryptsetup/cryptsetup.c

```diff
@@ -747,11 +747,13 @@ static int attach_luks_or_plain_or_bitlk_by_fido2(
 
         if (arg_fido2_cid) {
                 if (!key_file && !key_data)
-                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "FIDO2 mode selected but no key file specified, refusing.");
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "FIDO2 CID mode selected but no keyfile specified, refusing.");
```

Comment:
hmm?

"cid" the fido credential ID. it's kinda the primary piece of information for FIDO2 assertion requests. If it's set we we assume the user wants FIDO2 unlocking, and wants to specify the parameters manually instead of us loading them from JSON. Thus we complain here if the user didn#t also specifiy a key file, because besides the cid the keyfile is the one data piece we really really need. For all other bits we have good fallbacks.

Thus, the log message talked about "FIDO2 mode" correctly. I mean, if you want to improve it, maybe say "FIDO2 mode with manual parameters" or so, as opposed to "FIDO2 mode with parameters loaded from LUKS2 JSON metadata".